src/main/webapp/app/shared/util/firebase/firebase-crud-store.ts

```diff
@@ -0,0 +1,120 @@
+import { IRootStore } from 'app/stores';
+import { Database, onValue, push, ref, remove, set, update } from 'firebase/database';
+import { action, autorun, makeObservable, observable } from 'mobx';
+import { notifyError } from 'app/oncokb-commons/components/util/NotificationUtils';
+import { Review } from '../../model/firebase/firebase.model';
+import { convertNestedObject, getValueByNestedKey } from './firebase-utils';
+
+export type RecursivePartial<T> = {
+  [P in keyof T]?: RecursivePartial<T[P]>;
+};
+
+/* eslint-disable @typescript-eslint/ban-types */
+export type RecursiveKeyOf<TObj extends object> = {
+  [TKey in keyof TObj & (string | number)]: TObj[TKey] extends any[]
+    ? `${TKey}`
+    : TObj[TKey] extends object
+    ? `${TKey}` | `${TKey}/${RecursiveKeyOf<TObj[TKey]>}`
+    : `${TKey}`;
+}[keyof TObj & (string | number)];
+
+export class FirebaseCrudStore<T> {
```

Comment:
I'm trying to follow based on how JHipster generated `base-crud-store.ts`, `base-store.ts`, etc, which are all located under the utils folder. 

I think it would make sense if we move all `stores` to `webapp/stores` then?

Reply:
I think they should all be included under the firebase folder. Let's discuss during the standup